jupyterhub/templates/image-puller/_daemonset-helper.yaml

```diff
@@ -68,6 +68,15 @@ spec:
             - -c
             - echo "Pulling complete"
         {{- end }}
+        {{- range $k, $v := .Values.singleuser.extraContainers }}
+        - name: image-pull-singleuser-extra-container-{{ $k }}
+          image: {{ $v.image }}
+          imagePullPolicy: IfNotPresent
+          command:
+            - /bin/sh
+            - -c
+            - echo "Pulling complete"
+        {{- end }}
```

Comment:
>
```yaml
image: {{ $v.image }}
```

I wonder about this. Can you add an demonstrative value for `singleuser.extraContainers` to the config we use to lint and verify consistency in the templates as well? It resides in tools/templates.

Reply:
Note that you can use the linter yourself without needing to use the CI by running it from tools/templates/lint-and-validate.py (if i recall the path to it correctly)

Reply:
I have added a new lint file in tools/templates. `singleuser.extraContainers` can be used as the same with `singleuser.initContainers` or `hub.extraContainers` :)